pkg/ccpatch/substitution/substitution.go

```diff
@@ -0,0 +1,32 @@
+package substitution
+
+import (
+	"strings"
+)
+
+// Substitution defines a substitution that can be used to replace
+// placeholders in the cloud-init configuration.
+type Substitution struct {
+	Type                 string            `json:"type" yaml:"type"` // Type of the substitution, ipv4Address, ipv6Address, etc
+	Key                  string            `json:"key" yaml:"key"`   // Name of the substitution to be replaced $ipv4Address for example
```

Comment:
Maybe would be more explicit to name it `placeHolderName` instead of the generic `Key`

Reply:
relates to: https://github.com/ionos-cloud/crossplane-provider-ionoscloud/pull/235#discussion_r1611062806

Reply:
Ok, will close this comment